### DISCUSSION TITLE
how to change the getResultList offset

### DISCUSSION THREAD

**Discussion by rrrokhtar** [mikro-orm/mikro-orm#5948] [Q&A]
Greetings. I am on mikro-orm/mysql
- I ran into an issue after having my query returning with its many to many relations when I do some ordersBy and limit-offset 

> Expression # 2 of ORDER BY clause is not in GROUP BY clause and contains nonaggregated column

the resolution of that was getting my results back using **getResultList** the point now, I can see that I can send limit parameter, but cannot find out how can I send offset. 

>    getResultList(limit?: number): Promise<Loaded<Entity, Hint>[]>;

Using .offset / .limit does not work throws same error above...

**Comment by rrrokhtar** [comment]
BTW that function executes query.all (I recommend making it a little efficient and adding offset)

**Comment by B4nan** [comment]
The limit parameter in `getResultList` is rather internal, you should paginate on SQL level, not in memory. If you want to offset the value in memory, just use the native `slice` method or similar.

Hard to help without seeing what you are actually doing, but I guess you should order by `min(the_problematic_column)` or similar to get around this.

**Comment by rrrokhtar** [comment]
Alright, thanks @B4nan 👍
